src/rdf4cpp/bnode_mngt/reference_backends/scope/ReferenceNodeScope.hpp

```diff
@@ -55,15 +29,55 @@ struct ReferenceNodeScope : INodeScope {
                                  std::string_view>
             handle_to_label_;
 
-    ReferenceNodeScope(ReferenceNodeScope const &other);
 public:
-    ReferenceNodeScope() = default;
-    ReferenceNodeScope(ReferenceNodeScope &&other) noexcept;
+    template<typename ...GenArgs>
+    ReferenceNodeScope(GenArgs &&...gen_args) noexcept(std::is_nothrow_constructible_v<G, decltype(std::forward<GenArgs>(gen_args))...>)
+        : gen_{std::forward<GenArgs>(gen_args)...} {
+    }
+
+    [[nodiscard]] G const &generator() const noexcept {
+        return gen_;
+    }
```

Comment:
is there a use case where we need to access the underlying generator?

Reply:
not sure tbh, just added it because it might be useful